ProjectLighthouse.Servers.Website/Pages/PasswordResetPage.cshtml.cs

```diff
@@ -19,8 +21,21 @@ public PasswordResetPage(Database database) : base(database)
     [UsedImplicitly]
     public async Task<IActionResult> OnPost(string password, string confirmPassword)
     {
-        User? user = this.Database.UserFromWebRequest(this.Request);
-        if (user == null) return this.Redirect("~/login");
+        User? user;
+        if (Request.Query.ContainsKey("token"))
+        {
+            user = this.Database.UserFromPasswordResetToken(Request.Query["token"][0]);
+            if (user == null)
+            {
+                this.Error = "Invalid password reset token.";
```

Comment:
Hmm, this could be a little more user-friendly. Perhaps something like this might work a little better?
```suggestion
                this.Error = "This password reset link either is invalid or has expired. Please try again.";
```